customer/src/main/java/no/unit/nva/customer/CustomerDoiHandler.java

```diff
@@ -1,26 +1,43 @@
 package no.unit.nva.customer;
 
+import static java.util.Objects.isNull;
+import static no.unit.nva.commons.json.JsonUtils.dtoObjectMapper;
 import static nva.commons.apigateway.AccessRight.EDIT_OWN_INSTITUTION_RESOURCES;
 import static nva.commons.apigateway.AccessRight.EDIT_OWN_INSTITUTION_USERS;
 import static nva.commons.core.attempt.Try.attempt;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import java.net.URI;
+import java.util.Arrays;
+import java.util.Map;
 import java.util.UUID;
+import java.util.stream.Collectors;
 import no.unit.nva.customer.exception.InputException;
-import no.unit.nva.customer.model.CustomerDto.DoiAgentDto;
+import no.unit.nva.customer.model.SecretManagerDoiAgentDao;
 import nva.commons.apigateway.ApiGatewayHandler;
 import nva.commons.apigateway.RequestInfo;
 import nva.commons.apigateway.exceptions.ForbiddenException;
+import nva.commons.core.JacocoGenerated;
 import nva.commons.core.attempt.Failure;
+import nva.commons.core.paths.UriWrapper;
+import nva.commons.secrets.SecretsReader;
 
-public abstract class CustomerDoiHandler<I> extends ApiGatewayHandler<I, DoiAgentDto> {
+public abstract class CustomerDoiHandler<I> extends ApiGatewayHandler<I, String> {
 
     public static final String IDENTIFIER = "identifier";
     public static final String IDENTIFIER_IS_NOT_A_VALID_UUID = "Identifier is not a valid UUID: ";
+    protected static final String SECRETS_KEY_AND_NAME = "dataCiteCustomerSecrets";
+    protected final SecretsReader secretsReader;
+    // TODO add this code in task #NP-27814
+    //protected final ObjectMapper mapperToJsonCompact = createJsonParser();
```

Comment:
should this be here? WiP?

Reply:
I went a little too fast, next qa is this change.